frontend/src/components/BulkAssignProcedure.vue

```diff
@@ -0,0 +1,93 @@
+<script setup lang="ts">
+import { useToast } from "primevue/usetoast"
+import { ref } from "vue"
+import AssignProcedure from "@/components/AssignProcedure.vue"
+import { InfoStatus } from "@/components/enumInfoStatus"
+import InfoModal from "@/components/InfoModal.vue"
+import DocumentUnitListEntry from "@/domain/documentUnitListEntry"
+import { Procedure } from "@/domain/procedure"
+import { PublicationState } from "@/domain/publicationStatus"
+import DocumentUnitService from "@/services/documentUnitService"
+
+const props = defineProps<{
+  documentationUnits: DocumentUnitListEntry[] | undefined
+}>()
+const emit =
+  defineEmits<
+    (
+      e: "updateSelectionErrors",
+      error: string | undefined,
+      documentationUnitIdsWithErrors: string[],
+    ) => void
+  >()
+
+const hasBulkAssignError = ref(false)
+const toast = useToast()
+
+const assignProcedure = async (procedure: Procedure) => {
+  hasBulkAssignError.value = false
+  if (!areSelectedDocUnitsValid()) {
+    return
+  }
+
+  const documentationUnitIds =
+    props.documentationUnits?.map((docUnit) => docUnit.uuid!) ?? []
+  const { error } = await DocumentUnitService.bulkAssignProcedure(
+    procedure.label,
+    documentationUnitIds,
+  )
+
+  if (!error) {
+    const isPlural = documentationUnitIds.length > 1
+    const verb = isPlural ? "en sind" : " ist"
+    toast.add({
+      severity: "success",
+      summary: "Hinzufügen erfolgreich",
+      detail: `Die Dokumentationseinheit${verb} jetzt im Vorgang ${procedure.label}.`,
+      life: 5_000,
+    })
+  }
+
+  hasBulkAssignError.value = !!error
+}
+
+const areSelectedDocUnitsValid = () => {
+  if (!props.documentationUnits || props.documentationUnits.length === 0) {
+    emit(
+      "updateSelectionErrors",
+      "Wählen Sie mindestens eine Dokumentationseinheit aus",
+      [],
+    )
+    return false
+  }
+  const pendingDocUnitIds = props.documentationUnits
+    .filter(
+      (docUnit) =>
+        docUnit.status?.publicationStatus ===
+        PublicationState.EXTERNAL_HANDOVER_PENDING,
+    )
+    .map((docUnit) => docUnit.uuid!)
+  if (pendingDocUnitIds.length > 0) {
+    emit(
+      "updateSelectionErrors",
+      "Nehmen Sie die Fremdanlage(n) an, um sie zu einem Vorgang hinzuzufügen",
+      pendingDocUnitIds,
+    )
+    return false
+  }
+  emit("updateSelectionErrors", undefined, [])
+  return true
+}
+</script>
+
+<template>
+  <InfoModal
```

Comment:
Vielleicht habe ich was übersehen, aber ich verstehe noch nicht ganz welche art von fehlermeldung wir hier nutzen wollen. Unser altes InfoModal? den [primevue toast](https://primevue.org/toast/)? Die [ris-ui message](https://digitalservicebund.github.io/ris-ui/?path=/docs/primevue-message--docs), wie in [diesem ticket ](https://digitalservicebund.atlassian.net/jira/software/c/projects/RISDEV/boards/7/backlog?selectedIssue=RISDEV-7663) beschrieben?

Reply:
Good point. Ich habe es gerade mit der <Message> ausprobiert, aber dann müssen wir noch das Icon tauschen und so weiter. Ich habe es jetzt mal wieder ausgebaut. Denke macht mehr Sinn, wenn man sich die Stellen zusammen anschaut und vielleicht ersetzen wir ja dann auch das InfoModal durch eine neue ris-backend Komponente, die dann eine gestylte <Message> ist. 🤷